File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] result = new int[nums.length];

        int leftProduct = 1;
        for (int idx = 0; idx < nums.length; idx++){
            result[idx] = leftProduct;
            leftProduct *= nums[idx];
        }

        int rightProduct = 1;
        for (int idx = nums.length - 1; idx > -1; idx--){
            result[idx] *= rightProduct;
            rightProduct *= nums[idx];
        }

        return result;
    }
}  

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
#include <iostream>
#include <vector>



class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> seen;
        for (int num : nums) seen.insert(num);

        int result = 0;

        for (int num : nums){
            if (seen.find(num - 1) == seen.end()){
                int length = 1;
                while (seen.find(num + length) != seen.end()) length++;
                result = max(result, length);
            }

        }
        return result;
    }
};
